plugin/src/com/microsoft/alm/plugin/external/exceptions/VisualStudioClientVersionException.java

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.external.exceptions;
+
+import com.microsoft.alm.plugin.external.models.ToolVersion;
+
+public class VisualStudioClientVersionException extends ToolException {
+    final ToolVersion versionFound;
+    final ToolVersion minimumVersion;
+
+    public VisualStudioClientVersionException(final ToolVersion versionFound, final ToolVersion minimumVersion) {
+        super(ToolException.KEY_TF_VS_MIN_VERSION_WARNING);
+        this.versionFound = versionFound;
+        this.minimumVersion = minimumVersion;
+    }
+
+    @Override
+    public String[] getMessageParameters() {
+        return new String[] {versionFound.toString(), minimumVersion.toString()};
```

Comment:
I don't like this method. It looks like magic in the call site.

Reply:
It is the usual thing we do in parametrized exception classes. E.g. see the existing `ToolVersionException`: https://github.com/microsoft/azure-devops-intellij/blob/e0e3f36144ba2993a45e00ac3ffc50e3d3539828/plugin/src/com/microsoft/alm/plugin/external/exceptions/ToolVersionException.java#L23-L25